crates/ruff_linter/src/rules/flynt/rules/static_join_to_fstring.rs

```diff
@@ -126,14 +128,31 @@ fn build_fstring(joiner: &str, joinees: &[Expr], flags: FStringFlags) -> Option<
         if !std::mem::take(&mut first) {
             f_string_elements.push(helpers::to_interpolated_string_literal_element(joiner));
         }
-        f_string_elements.push(helpers::to_interpolated_string_element(expr)?);
+        let element = helpers::to_interpolated_string_element(expr)?;
+        if let ast::InterpolatedStringElement::Literal(ast::InterpolatedStringLiteralElement {
+            value,
+            ..
+        }) = &element
+        {
+            has_single_quote |= value.contains('\'');
+            has_double_quote |= value.contains('"');
+        }
+        f_string_elements.push(element);
     }
 
+    let quote = flags.quote_style();
+    let adjusted_quote = match quote {
+        Quote::Single if has_single_quote && !has_double_quote => quote.opposite(),
+        Quote::Double if has_double_quote && !has_single_quote => quote.opposite(),
+        _ if has_double_quote && has_single_quote => return None,
```

Comment:
If the content includes both single quote and double quote, ruff skips FLY002 for now.

I will improve this logic in a separate pull request as we discussed here.

https://github.com/astral-sh/ruff/pull/20197#pullrequestreview-3240917942